semgrep-core/src/core/il/IL.ml

```diff
@@ -205,18 +189,18 @@ and var_special = This | Super | Self | Parent
 (* Expression *)
 (*****************************************************************************)
 
+(*s: type [[IL.exp]] *)
 (* We use 'exp' instead of 'expr' to accentuate the difference
  * with AST_generic.expr.
  * Here 'exp' does not contain any side effect!
+ * todo: etype: typ;
  *)
-(*s: type [[IL.exp]] *)
-and exp = { e : exp_kind; (* todo: etype: typ; *)
-                          eorig : G.expr }
+and exp = { e : exp_kind; eorig : G.expr }
 
 (*e: type [[IL.exp]] *)
 (*s: type [[IL.exp_kind]] *)
 and exp_kind =
-  | Lvalue of lval (* lvalue used in a rvalue context *)
```

Comment:
I used to call this `Fetch` (since you're fetching the value contained in the lvalue). I still think `Lvalue` was more appropriate than `Rvalue`, because basically any `exp_kind` is an r-value, and `Lvalue` made clear that it is the r-value obtained from an l-value.

Reply:
Ok, I'll rename Fetch. I like Fetch.